application/client/src/app/ui/tabs/sources/multiplefiles/component.ts

```diff
@@ -0,0 +1,115 @@
+import { Component, AfterContentInit, Input, HostListener } from '@angular/core';
+import { Ilc, IlcInterface } from '@env/decorators/component';
+import { Initial } from '@env/decorators/initial';
+import { FileType, File } from '@platform/types/files';
+import { Action } from '../common/actions/action';
+import { FileHolder } from './file.holder';
+import { bytesToStr } from '@env/str';
+import { Holder } from '@module/matcher';
+
+@Component({
+    selector: 'app-tabs-source-multiple-files',
+    templateUrl: './template.html',
+    styleUrls: ['./styles.less'],
+})
+@Initial()
+@Ilc()
+export class TabSourceMultipleFiles extends Holder implements AfterContentInit {
+    @Input() files!: File[];
+
+    @HostListener('window:keydown', ['$event'])
+    handleKeyDown(event: KeyboardEvent) {
+        this.matcher.search(event.key);
+        this.ngFiles = this._files
+            .sort((a: FileHolder, b: FileHolder) => b.getScore() - a.getScore())
+            .filter((file: FileHolder) => file.getScore() > 0);
+    }
+
+    public action: Action = new Action();
+    public ngCompatible: boolean = false;
+    public ngSelectedCount: number = 0;
+    public ngSelectedSize: string = '0 b';
+    public ngDestination: string = '';
+    public ngFiles: FileHolder[] = [];
+
+    private _files: FileHolder[] = [];
```

Comment:
all these stuff can be grouped into State, which you can use after to save and load (to support tab's switching). In layout you also can directly access to state, like `{{state.size}}` instead `ngSelectedSize`